### PR TITLE
cleanup trigger data

### PR DESCRIPTION
**Pull request by emptyrivers**
This pull request introduces an internalVersion change, and moves all trigger data into a single table, for ease of manipulation. There should be no change in behavior that is apparent to the end user.

### CODE REVIEW

**Comment by emptyrivers** [comment]
Existing auras have their data migrated correctly, and behave as expected in normal use. 

Options interaction and importing still require testing.

**Comment by emptyrivers** [comment]
Everything seems to be in order with the latest changes on this pull request. However, since it involves changes that intersect with #664 and #660, I believe that it would be better to wait until after work on supercharged templates is complete.

File: WeakAuras/WeakAuras.lua [hunk 1]
```diff
@@ -2532,6 +2532,42 @@ function WeakAuras.Modernize(data)
...
+    if activeTriggerMode ~= WeakAuras.trigger_modes.first_active then
+      activeTriggerMode = activeTriggerMode + 1
+    end
+    data.triggers.activeTriggerMode = activeTriggerMode
+    data.activeTriggerMode = nil
+    data.triggers.disjunctive = data.disjunctive
+    data.disjunctive = nil
+    -- migrate condition trigger references
+    for _, condition in pairs(data.conditions) do
+      if condition.check.trigger and condition.check.trigger ~= -1 then
```
Comment (InfusOnWoW) [inline WeakAuras/WeakAuras.lua]: That isn't enough anymore with AND/OR conditions.

File: WeakAuras/WeakAuras.lua [hunk 2]
```diff
@@ -2678,8 +2712,8 @@ local function pAdd(data)
     db.displays[id] = data;
     WeakAuras.SetRegion(data);
   else
-    if (not data.activeTriggerMode or data.activeTriggerMode >= data.numTriggers) then
-      data.activeTriggerMode = WeakAuras.trigger_modes.first_active;
+    if (not data.triggers.activeTriggerMode or data.triggers.activeTriggerMode >= #data.triggers) then
```
Comment (InfusOnWoW) [inline WeakAuras/WeakAuras.lua]: \>= is now wrong here.

File: WeakAuras/WeakAuras.lua [hunk 3]
```diff
@@ -3504,7 +3534,7 @@ local function wrapTriggerSystemFunction(functionName, mode)
     else -- triggernum >= 0
       local triggerSystem = WeakAuras.GetTriggerSystem(data, triggernum);
       if (not triggerSystem) then
-        return false;
+        return false, print('no trigger system when checking',data.id,triggernum);
```
Comment (InfusOnWoW) [inline WeakAuras/WeakAuras.lua]: spaces after comma
Also no reason to not do that in two lines

File: WeakAurasOptions/ConditionOptions.lua [hunk 4]
```diff
@@ -1302,7 +1302,7 @@ local function createConditionTemplates(data)
   conditionTemplates.display = {};
 
   local index = 1;
-  for triggernum = -1, numTriggers - 1 do
+  for triggernum = -1, numTriggers do
```
Comment (InfusOnWoW) [inline WeakAurasOptions/ConditionOptions.lua]: This needs to skip 0 now.

File: WeakAuras/Transmission.lua [hunk 5]
```diff
@@ -1033,11 +1027,8 @@ end
 
 
 local function scamCheck(codes, data)
-  checkTrigger(codes, L["%s - 1. Trigger"]:format(data.id), data.trigger, data.untrigger);
-  if (data.additional_triggers) then
-    for i, v in ipairs(data.additional_triggers) do
-      checkTrigger(codes, L["%s - %i. Trigger"]:format(data.id, i+1), v.trigger, v.untrigger);
-    end
+  for i, v in ipairs(data.triggers) do
+    checkTrigger(codes, L["%s - %i. Trigger"]:format(data.id, i+1), v.trigger, v.untrigger);
```
Comment (InfusOnWoW) [inline WeakAuras/Transmission.lua]: i+1 is wrong now.

File: WeakAuras/WeakAuras.lua [hunk 6]
```diff
@@ -2587,6 +2587,42 @@ function WeakAuras.Modernize(data)
...
+    local activeTriggerMode = data.activeTriggerMode or WeakAuras.trigger_modes.first_active
+    if activeTriggerMode ~= WeakAuras.trigger_modes.first_active then
+      activeTriggerMode = activeTriggerMode + 1
+    end
+    data.triggers.activeTriggerMode = activeTriggerMode
+    data.activeTriggerMode = nil
+    data.triggers.disjunctive = data.disjunctive
+    data.disjunctive = nil
+    -- migrate condition trigger references
+    for _, condition in pairs(data.conditions) do
```
Comment (InfusOnWoW) [inline WeakAuras/WeakAuras.lua]: This still doesn't work

**Comment by InfusOnWoW** [comment]
As far as I can tell, this is now bug free. It will break some users cuistom code, but I think not too many, and there's indeed a good improvement overall.

**Pull request merged**